test/unit/polling_srv_records_for_mongos_discovery.prose.test.ts

```diff
@@ -0,0 +1,258 @@
+      { name: 'localhost.test.mock.test.build.10gen.cc', port: 2020, weight: 0, priority: 0 }
+    ];
+
+    makeStubs({ initialRecords, replacementRecords });
+
+    client = new MongoClient(SRV_CONNECTION_STRING, {
+      tls: false,
+      srvMaxHosts: 2,
+      serverSelectionTimeoutMS: 5000
+    });
+    await client.connect();
+
+    const selectedHosts = client.topology.s.seedlist;
+    expect(selectedHosts).to.have.lengthOf(2);
+    expect(selectedHosts.map(({ host }) => host)).to.deep.equal(
+      initialRecords.map(({ name }) => name)
+    );
+
+    clock.tick(2 * client.topology.s.srvPoller.rescanSrvIntervalMS);
+    await processTick();
+
+    const polledServerAddresses = Array.from(client.topology.description.servers.keys());
+    polledServerAddresses.sort();
+    expect(polledServerAddresses).to.deep.equal(
+      replacementRecords.map(({ name, port }) => `${name}:${port}`)
+    );
+  });
+
+  it('12 - New DNS records are randomly selected (srvMaxHosts > 0)', async () => {
+    const replacementRecords = [
+      { name: 'localhost.test.mock.test.build.10gen.cc', port: 2017, weight: 0, priority: 0 },
+      { name: 'localhost.test.mock.test.build.10gen.cc', port: 2019, weight: 0, priority: 0 },
+      { name: 'localhost.test.mock.test.build.10gen.cc', port: 2020, weight: 0, priority: 0 }
+    ];
+
+    makeStubs({ initialRecords, replacementRecords });
+
+    client = new MongoClient(SRV_CONNECTION_STRING, {
+      tls: false,
+      srvMaxHosts: 2,
+      serverSelectionTimeoutMS: 5000
+    });
+    await client.connect();
+
+    const selectedHosts = client.topology.s.seedlist;
+    expect(selectedHosts).to.have.lengthOf(2);
+    expect(selectedHosts.map(({ host }) => host)).to.deep.equal(
+      initialRecords.map(({ name }) => name)
+    );
+
+    clock.tick(2 * client.topology.s.srvPoller.rescanSrvIntervalMS);
+    await processTick();
+
+    const polledServerAddresses = Array.from(client.topology.description.servers.keys());
+    polledServerAddresses.sort();
+    // Only two addresses, one should remain the original 2017,
+    // while the other will be one of 2019 or 2020
+    expect(polledServerAddresses).to.have.lengthOf(2);
+    expect(polledServerAddresses.find(addr => addr.includes('2017'))).to.be.a('string');
+    expect(polledServerAddresses).satisfies(
+      addresses =>
+        // If you want proof, comment one of these conditions out, and run the test a few times
+        // you should see it pass and fail at random
+        typeof addresses.find(addr => addr.includes('2019')) === 'string' ||
+        typeof addresses.find(addr => addr.includes('2020')) === 'string'
+    );
+  });
+
+  it('13 - DNS record with custom service name can be found', async () => {
+    client = new MongoClient(SRV_CONNECTION_STRING, {
+      tls: false,
+      srvServiceName: 'myFancySrvServiceName',
+      serverSelectionTimeoutMS: 5000
+    });
+
+    makeStubs({ srvServiceName: 'myFancySrvServiceName' });
```

Comment:
I think it would be easy enough to make this match the actual spec test case - specify one initial record on port 2017, then return two records for ports 2019 and 2020, so why not do that?